### PR TITLE
Add OTLP logs to calendar-rest-go

### PR DESCRIPTION
**Pull request by liustanley**
### What does this PR do?

Adds logs over OTLP gRPC to the Go Calendar example. Also updates the timeout in the server to avoid errors.

I override the existing `logger` when adding the loggerProvider because we use the initial logger in the process of creating the loggerProvider. I'm not sure if there's a better way to set this up.

### Motivation

I plan to use this example for end to end pipeline tests in datadog-agent for traces, metrics, and logs.

### CODE REVIEW

**Comment by jackgopack4** [comment]
will overriding the logger cause us to lose the logs that are sent if there are errors initializing components, like Tracer provider?
Are we allowed to have a temporary logger to use before setup instead of overwriting the value of logger?

File: apps/rest-services/golang/calendar/main.go [hunk 1]
```diff
@@ -110,6 +113,20 @@ func initMetricProvider(res *resource.Resource) (*metric.MeterProvider, error) {
...
+
+	loggerProvider := log.NewLoggerProvider(
+		log.WithProcessor(log.NewBatchProcessor(otlpexp)),
+		log.WithResource(res),
+	)
+	return loggerProvider, nil
+}
+
 func setupHandlers(server *Server) *http.ServeMux {
 	mux := http.NewServeMux()
```
Comment (jackgopack4) [inline apps/rest-services/golang/calendar/main.go:132]: maybe we should get rid of this reference to rolldice on line 134

**Comment by jade-guiton-dd** [comment]
@jackgopack4 It looks to me like any errors during initialization of the tracer/metrics provider should be sent through the OTel-enabled logger. I think this 2 step approach where an initial stderr-only logger is replaced by an OTel-enabled one makes sense.

However, I wonder if it may be useful to still log things to stderr as well? I also think keeping the original logger config would be good. Maybe something like:
```go
otelCore := otelzap.NewCore(service, otelzap.WithLoggerProvider(loggerProvider))
logger = logger.WithOptions(
  zap.WrapCore(func(core zapcore.Core) zapcore.Core {
    return zapcore.NewTee(core, otelCore)
  }),
  zap.Fields(zap.String("service", service)),
)
```
I don't have any experience with zap, so no idea if this is idiomatic or not.

**Comment by liustanley** [comment]
@jade-guiton-dd Good callout, I updated my PR to include the stdout exporter in the loggerprovider which should be the idiomatic way to do this.

**Comment by jade-guiton-dd** [comment]
That is definitely simpler! Note that this erases the configuration of the original logger (ie. the options set by `NewDevelopment`), which currently seems to consist of `ErrorOutput(Open("stderr")), Development(), AddCaller(), AddStacktrace(WarnLevel)`. I'm not sure how important that is though, so I won't block you on that.

**Review by jade-guiton-dd — APPROVED** [review]

**Pull request merged**